kie-spring/src/test/java/org/kie/spring/tests/KieSpringScopeTest.java

```diff
@@ -88,6 +89,30 @@ public void testStatefulPrototypeKieSession() throws Exception {
 
         assertNotEquals(ksession.hashCode(), anotherKsession.hashCode());
     }
+    @Test
+    public void testConcurrentlyGetStatefulPrototypeKieSession() throws Exception {
+        final int nThreads = 10;
+        ExecutorService executorService = Executors.newFixedThreadPool(nThreads);
+        final ConcurrentHashMap ksessionMap = new ConcurrentHashMap();
+
+        CountDownLatch latch = new CountDownLatch(nThreads);
+
+        for (int i = 0; i < nThreads; i++) {
+            executorService.submit(new Runnable() {
+                @Override
+                public void run() {
+                    KieSession ksession = (KieSession) context.getBean("statefulPrototypeSession");
+                    Object put = ksessionMap.put(ksession.hashCode(), new Object());
```

Comment:
The fix is ok but using the hashCode() as unique key of a ksession isn't. I know this is uncommon but it's not impossible that 2 distinct ksessions have the same hashcode. Please replace ksession.hashCode() with ksession.getIdentifier()